### ISSUE TITLE
Build system support for first-class Python plugins

### ISSUE THREAD

**Issue by codebrainz** [geany/geany-plugins#297]
With the new proxy plugin support in Geany and eventual update of GeanyPy to use it, we should consider adding support to the build system(s) to allow first-class Python plugins.

A few things that should be considered initially:

- GeanyPy plugins should force `--enable-geanypy`. Or however the dependency is best cared for.
- They should probably be compiled to `.pyc` files on installation, similar to GeanyPy's `.py` files.
- Whether we should have each their directory in `$topsrcdir`, or just install all the plugins found in GeanyPy's `plugins` directory.
- Other stuff I'm sure...

**Comment by kugel-** [comment]
I agree. Regarding the location, I would rather not place them under geanypy but instead at the top-level just like other plugins. They are also gonna need NEWS, README, etc like everyone else.

**Comment by elextr** [comment]
Agree with @kugel- the Python plugins should be top level and have the usual boilerplate.  The user really should not care about what language the plugin was written in.

As for pycompiling them, IIUC the `.pyc` is only used if the Python being used is *exactly* the same version, but it doesn't hurt if it isn't, Python will just recompile it.  So I don't care either way.
